### PR TITLE
Fix for blank TimeStampField & small improvements

### PR DESCRIPTION
**Pull request by wuttem**
Thanks for creating this library!

I found a small bug when loading TimeStampField from database with blank attribute.
I also added the possibility to use callable defaults. This makes it possible to use
`default=datetime.datetime.utcnow`

Also I had to unpin the versions in the requirements.txt to use it in a project.

Not sure if you are interested in any of these changes - feel free to use or reject.

### CODE REVIEW

File: matchbox/models/field_validator.py [hunk 1]
```diff
@@ -26,7 +26,11 @@ def validate(self, f_name, value):
                 )
 
         if self.default is not None and value is None:
-            value = self.default
+            # check for factory
```
Comment (gameboy86) [inline matchbox/models/field_validator.py:29]: Please add to README information about this functionality

File: matchbox/models/fields.py [hunk 2]
```diff
@@ -198,7 +201,11 @@ def lookup_value(self, lookup_type, value):
         return self.db_value(value)
 
     def db_value(self, value):
-        if not issubclass(value.__class__, self.ref_model):
+        if hasattr(value, "full_collection_name") and hasattr(value, "id"):
```
Comment (gameboy86) [inline matchbox/models/fields.py]: I don't understand why we want to do that ? We specify on model level exactly which model can by set as referrer. If you need this functionality better will be create specify field for this. What do you think ?

**Comment by gameboy86** [comment]
Hi thanks for your work. Of Course i will merge your changes, but first look to review,  I wrote 2 questions.

File: matchbox/models/fields.py (hunk 2 above)
Comment (wuttem) [reply matchbox/models/fields.py]: This was very specific for trying tomaking lazy loading reference fields work ... Probably doesnt make sense for someone else...
removed it from the PR

**Pull request merged**